sklearnex/basic_statistics/incremental_basic_statistics.py

```diff
@@ -103,6 +103,12 @@ class IncrementalBasicStatistics(IntelEstimator, BaseEstimator):
     ----
     Attribute exists only if corresponding result option has been provided.
 
+    Note
+    ----
+    Since finalize_fit can't be dispatched without directly provided queue
```

Comment:
Thanks for adding this note. But it looks out of context without having seen the PR before - e.g. it doesn't quite convey the idea that something happens during calls to pickle, and that this something is for device data or sycl queues.

Perhaps you could rephrase it along the lines of "serializing instances of this class fitted to sycl data will trigger a forced finalization of calculations" or something similar, otherwise it's not very clear for a final user what all of this means.